src/app/routes/routes.ts

```diff
@@ -15,37 +15,36 @@ type RouteConfig = Record<RoutablePages, Omit<IRoute, "name">>;
 export type RoutePageMap = Record<RoutablePages, ComponentClass>;
 type RouteNavigate = Record<RoutablePages, (...params: any[]) => Action>;
 
-function getRoutes(routeConfig: RouteConfig): Record<RoutablePages, IRoute> {
-  return Object.keys(routeConfig)
-    .map((key) => ({
-      name: key,
-      path: routeConfig[key].path
-    }))
-    .reduce(
-      (a, c) => {
-        a[c.name] = c;
-        return a;
-      },
-      {} as any
-    );
-}
-
-function getNavigateAction<T extends {[key: string]: any}>(routeName: RoutablePages, params?: T): Action {
-  return actions.navigateTo(routeName, params);
-}
-
 const config: RouteConfig = {
   aboutPage: {path: "/about"},
   counterPage: {path: "/counter"},
   homePage: {path: "/"},
   starsPage: {path: "/stars"}
 };
 
-export const routes = getRoutes(config);
+let cachedRoutes: Record<RoutablePages, IRoute> = null;
+let cachedBaseUrl = "";
+
+export function getRoutes(baseUrl: string = ""): Record<RoutablePages, IRoute> {
+  if (!cachedRoutes || cachedBaseUrl !== baseUrl) {
```

Comment:
actually I'd have prefer not to cache here, but when we consume these, do a const `routes = getRoutes()` cacheBaseUrl is acting like a global (but not really, global to this file I mean)